app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -499,39 +501,39 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
 
                 override fun onPermissionRequest(request: PermissionRequest?) {
                     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                        val alreadyGranted = ArrayList<String>()
+                        val toBeGranted = ArrayList<String>()
                         request?.resources?.forEach {
                             if (it == PermissionRequest.RESOURCE_VIDEO_CAPTURE) {
                                 if (ActivityCompat.checkSelfPermission(
                                         context,
                                         android.Manifest.permission.CAMERA
                                     ) == PackageManager.PERMISSION_GRANTED
                                 ) {
-                                    request.grant(arrayOf(it))
+                                    alreadyGranted.add(it)
                                 } else {
-                                    requestPermission.launch(android.Manifest.permission.CAMERA)
+                                    toBeGranted.add(android.Manifest.permission.CAMERA)
                                 }
                             } else if (it == PermissionRequest.RESOURCE_AUDIO_CAPTURE) {
                                 if (ActivityCompat.checkSelfPermission(
                                         context,
                                         android.Manifest.permission.RECORD_AUDIO
                                     ) == PackageManager.PERMISSION_GRANTED
                                 ) {
-                                    request.grant(arrayOf(it))
+                                    alreadyGranted.add(it)
                                 } else {
-                                    requestPermission.launch(
-                                        android.Manifest.permission.RECORD_AUDIO
-                                    )
-                                }
-                            } else if (it == PermissionRequest.RESOURCE_AUDIO_CAPTURE) {
-                                if (ActivityCompat.checkSelfPermission(
-                                        context,
-                                        android.Manifest.permission.MODIFY_AUDIO_SETTINGS
```

Comment:
This actual permission is separate from the one above, I think we still need to keep it as the actual permission check is different. All 3 permissions are required as they were previously tested by users in the past. Different devices have different purposes.

Reply:
Dear @dshokouhi,
thank you for reviewing this PR.

As you can see from the original code, the _else if_ branch that you referenced was never reachable because the condition was exactly the same as the one before it:

line 513:
`} else if (it == PermissionRequest.RESOURCE_AUDIO_CAPTURE) {`

line 525:
`} else if (it == PermissionRequest.RESOURCE_AUDIO_CAPTURE) {`

Because these are _else if_ and not multiple _if_s, the second branch was completely unused.

What do you think?

Reply:
Ah that is a good catch indeed! In this case instead of removing that specific permission lets fix that bug here too if possible. Thank you for pointing that out.

Reply:
@dshokouhi you are welcome!
I checked Android documentation about PermissionRequest (https://developer.android.com/reference/android/webkit/PermissionRequest#constants_1) and only possible values are:

- RESOURCE_AUDIO_CAPTURE (already handled)
- RESOURCE_VIDEO_CAPTURE (already handled)
- RESOURCE_MIDI_SYSEX (unrelated, sysex messages from MIDI devices)
- RESOURCE_PROTECTED_MEDIA_ID (unrelated, needed to generate licenses for protected media)

So I don't see any use case that could need the `android.Manifest.permission.MODIFY_AUDIO_SETTINGS` permission.
I could add it to the `RESOURCE_AUDIO_CAPTURE` case but I don't think this is needed and it could result in asking unnecessary permissions to the user.

As it was never used in the past I suggest to just remove it, maybe it was a leftover from tests during development, but if you prefer to add it to the `RESOURCE_AUDIO_CAPTURE` case I will edit the PR.

Thank you

Reply:
Thats strange. Looking at the original PR #1850 I can see that this code was always there but looking at the request the user pointed out the current version of the app (at the time of the request) did not work for them and the debug APK which contained that specific permission was there and that it was now working for their setup. I think we should still keep it there especially considering the error message in the request that calls out this specific permission. I don't believe this permission is auto granted once added to the manifest so I am confused as to how that PR fixed the users issue 🤔  Personally speaking I don't think it hurts to ask for more permissions especially if something is going to require it. The main thing to consider here is that while it may have been broken for some devices or users it wasn't broken for all.

Reply:
@dshokouhi I think that what fixed the issue of the user was adding the `<uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS" />` to the AndroidManifest.xml.
Do you agree that this code was never reached due to the same else if condition?

Reply:
Yea that was probably it, that permission may have been granted alongside record audio.

Sorry for the confusion had to look up the original PR since I remembered the request 😅

Reply:
Ok I can confirm this. Take a look at Android permission list (https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/res/AndroidManifest.xml).
`android.permission.MODIFY_AUDIO_SETTINGS` has `android:protectionLevel="normal"` so it doesn't need approval from the user, just the declaration in the manifest.
(as opposed to `android.permission.RECORD_AUDIO` that has `android:protectionLevel="dangerous|instant"`)
So what fixed the issue of the user was adding the permission to the manifest. It can be safely removed from the code

Reply:
Thank you for clarifying that :) I also see that Android Studio properly marks that condition to notify users of that mistake 😓

Reply:
You are right, didn't noticed! 😆